testcases/OpTestKexec.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+# IBM_PROLOG_BEGIN_TAG
+# This is an automatically generated prolog.
+#
+# $Source: op-test-framework/testcases/OpTestKexec.py $
+#
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2018-2019
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+#
+
+'''
+OpTestKexec
+---------
+
+This test is to validate kexec commands
+'''
+
+import os
+import pexpect
+import unittest
+import time
+import OpTestLogger
+
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common.OpTestError import OpTestError
+
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+class OpTestKexec(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.c = self.cv_SYSTEM.console
+        self.kernel_image = conf.args.kernel_image
+        self.initrd_image = conf.args.initrd_image
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in (res[0] or res[1]):
+            self.distro = "ubuntu"
+        elif 'Red Hat' in (res[0] or res[1]):
+            self.distro = 'rhel'
+        elif 'SLES' in (res[0] or res[1]):
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported only on ubuntu, sles and rhel")
+
+ 
+    def kexec_load(self):
+        """
+        This function will test the kexec load
+        """
+
+        try:
+            self.c.run_command("kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image))
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "0":
+                log.info("New kernel not loaded- Failed with kexec_loaded_status = \"%s\"", kexec_loaded_status[0])
```

Comment:
Instead of printing kexec_loaded_status value it will be better to print the kernel image name. 
Since kexec_loaded_status value will be zero in case of failure, the error message with status = 0 may be confusing. 

log.info("Failed to load kernel image = %s", self.kernel_image)

Reply:
Done